File: Low Level Design (LLD)/DesignPatterns/3. Structural/Visitor/Java/Main.java
public class Main {
    public static void main(String[] args) {
        Directory root = new Directory("Root");
        File file1 = new File("Resume.pdf");
        File file2 = new File("Photo.png");

        Directory docs = new Directory("Documents");
        docs.add(file1);

        Directory images = new Directory("Images");
        images.add(file2);

        root.add(docs);
        root.add(images);

        Visitor visitor = new NamePrinterVisitor();
        root.accept(visitor);
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Visitor/Cpp/main.cpp
#include "include/Directory.h"
#include "include/File.h"
#include "include/NamePrinterVisitor.h"

int main() {
    Directory* root = new Directory("Root");

    File* file1 = new File("Resume.pdf");
    File* file2 = new File("Photo.png");

    Directory* docs = new Directory("Documents");
    docs->add(file1);

    Directory* images = new Directory("Images");
    images->add(file2);

    root->add(docs);
    root->add(images);

    NamePrinterVisitor* visitor = new NamePrinterVisitor();
    root->accept(visitor);

    delete visitor;
    delete root;

    return 0;
}
